domain/application/service/application.go

```diff
@@ -263,11 +273,76 @@ func (s *ApplicationService) AddUnits(ctx context.Context, name string, units ..
 }
 
 // DeleteUnit deletes the specified unit.
+// TODO(units) - rework when dual write is refactored
+// This method is called (mostly during cleanup) after a unit
+// has been removed from mongo. The mongo calls are
+// DestroyMaybeRemove, DestroyWithForce, RemoveWithForce.
```

Comment:
Comment the possible errors to return.

Reply:
There are no domain errors returned. If the unit doesn't exit, it's a no op.